docs/optimization.rst

```diff
@@ -0,0 +1,37 @@
+Memory optimization
+===================
+
+Executable code size
+--------------------
+In order to manage the executable code size, the library can be compiled enabling or disabling several profiles.
+To add or remove a profile from the library, edit the ``client.config`` file.
+More information can be found at: :ref:`micro_xrce_dds_client_label`.
+
+Runtime size
+------------
+The client is static memory free.
+This implies that all memory charge depends of how the stack grows during the execution.
+To control the stack, there are some values that can be modified and impact to a large degree:
+
+Stream buffers
+~~~~~~~~~~~~~~
+1. Make sure you defined only the streams that will be used.
```

Comment:
It would be useful a link to the section where stream concept is explained.

Reply:
I think that would be useful too :smiley:, but we still do not have a dedicated part for it. Comming soon.
Besides, the optimization part is only useful for advance users, and this users must know already the stream concepts